### PR TITLE
crunch: new recipe

### PR DESCRIPTION
**Pull request by jsteinaker**
A wordlist generator.
Not sure about the ARCHITECTURES, should I disable x86_gcc2 and enable x86 in SECONDARY_ARCHITECTURES? This should compile with every compiler under the sun.

### CODE REVIEW

**Comment by jmairboeck** [comment]
If it builds without patching gcc2, I think this is fine. No need to make the recipe more complicated than it needs to be.

**Comment by jsteinaker** [comment]
Thanks for testing @Begasus. I guess commandSuffix it's not needed since we're not building for x86_gcc2 anyway?

File: app-misc/crunch/crunch-3.6.recipe [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/crunch-wordlist/files/crunch-wordlist/crunch-$portVersion.tgz"
+CHECKSUM_SHA256="6a8f6c3c7410cc1930e6854d1dadc6691bfef138760509b33722ff2de133fe55"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	crunch$secondaryArchSuffix = $portVersion
+	cmd:crunch$secondaryArchSuffix
```
Comment (Begasus) [inline app-misc/crunch/crunch-3.6.recipe:17]: For your latest comment, no commandSuffix needed, but also no $secondaryArchSuffix, the binary will be installed in $prefix/bin, so:
```suggestion
	cmd:crunch
```

File: app-misc/crunch/crunch-3.6.recipe [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"
+REQUIRES="
+	haiku
```
Comment (Begasus) [inline app-misc/crunch/crunch-3.6.recipe:20]: Haiku should also be secondary architecture :)

```suggestion
	haiku$secondaryArchSuffix
```

File: app-misc/crunch/crunch-3.6.recipe [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"
+
+BUILD_REQUIRES=""
+BUILD_PREREQUIRES="
+	haiku_devel
```
Comment (Begasus) [inline app-misc/crunch/crunch-3.6.recipe:25]:
```suggestion
	haiku${secondaryArchSuffix}_devel
```

**Review by Begasus — CHANGES_REQUESTED** [review]

File: app-misc/crunch/crunch-3.6.recipe (hunk 3 above)
Comment (Begasus) [reply app-misc/crunch/crunch-3.6.recipe:25]: Actually haiku${secondaryArchSuffix}_devel should go to BUILD_REQUIRES instead of BUILD_PREREQUIRES. :)

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks!

**Pull request merged**